mat64/io_test.go

```diff
@@ -141,7 +237,29 @@ func TestDenseIORoundTrip(t *testing.T) {
 		}
 
 		if !test.eq(&got, test.want) {
-			t.Errorf("r/w test #%d failed\nwant=%#v\n got=%#v\n", i, test.want, &got)
+			t.Errorf("r/w test #%d failed\n got=%#v\nwant=%#v\n", i, &got, test.want)
+		}
+
+		wbuf := new(bytes.Buffer)
+		_, err = test.want.MarshalBinaryTo(wbuf)
+		if err != nil {
+			t.Errorf("error encoding test #%d: %v\n", i, err)
+		}
+
+		if !bytes.Equal(buf, wbuf.Bytes()) {
+			t.Errorf("encoding via MarshalBinary and MarshalBinaryTo differ:\nwith-stream: %q\n  no-stream: %q\n",
+				i, wbuf.Bytes(), buf,
+			)
+		}
+
+		var wgot Dense
+		_, err = wgot.UnmarshalBinaryFrom(wbuf)
```

Comment:
do we need it?
we test all these properties:
buf == wbuf
wbuf -> wgot and wgot == test.want
buf->got and got == test.want

I don't really see what else could be tested.